adapters/oath/oath.go

```diff
@@ -0,0 +1,178 @@
+
+	for _, imp := range request.Imp {
+		//Oath supports only banner and video impressions as of now
+		if imp.Banner != nil {
+			bannerImps = append(bannerImps, imp)
+		} else if imp.Video != nil {
+			videoImps = append(videoImps, imp)
+		} else {
+			err := &adapters.BadInputError{
+				Message: fmt.Sprintf("Oath only supports banner and video imps. Ignoring imp id=%s", imp.ID),
+			}
+			errs = append(errs, err)
+		}
+	}
+
+	if len(bannerImps) <= 0 && len(videoImps) <= 0 {
+		err := &adapters.BadInputError{
+			Message: fmt.Sprintf("No valid impression in the bid request"),
+		}
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	reqJSON, err := json.Marshal(request)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+	errors := make([]error, 0, 1)
+
+	var bidderExt adapters.ExtImpBidder
+	err = json.Unmarshal(request.Imp[0].Ext, &bidderExt)
+
+	if err != nil {
+		err = &adapters.BadInputError{
+			Message: fmt.Sprintf("ext.bidder not provided"),
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	var oathExt openrtb_ext.ExtImpOath
+	err = json.Unmarshal(bidderExt.Bidder, &oathExt)
+	if err != nil {
+		err = &adapters.BadInputError{
+			Message: fmt.Sprintf("ext.bidder.publisherName not provided"),
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	if oathExt.PublisherName == "" {
+		err = &adapters.BadInputError{
+			Message: "publisherName is empty",
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
+	thisURI := uri
+	thisURI = thisURI + "?publisher=" + oathExt.PublisherName
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+
+	if request.Device != nil {
+		addHeaderIfNonEmpty(headers, "User-Agent", request.Device.UA)
+		addHeaderIfNonEmpty(headers, "X-Forwarded-For", request.Device.IP)
+		addHeaderIfNonEmpty(headers, "Accept-Language", request.Device.Language)
+		addHeaderIfNonEmpty(headers, "DNT", strconv.Itoa(int(request.Device.DNT)))
+	}
+
+	return []*adapters.RequestData{{
+		Method:  "POST",
+		Uri:     thisURI,
+		Body:    reqJSON,
+		Headers: headers,
+	}}, errors
+}
+
+func (a *OathAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&adapters.BadInputError{
+			Message: fmt.Sprintf("Unexpected status code: %d. ", response.StatusCode),
+		}}
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{fmt.Errorf("Unexpected status code: %d. ", response.StatusCode)}
+	}
+
+	var bidResp openrtb.BidResponse
+	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(5)
```

Comment:
Will you ever find more than one SeatBid in your response from Oath? if not, then
`adapters.NewBidderResponseWithBidsCapacity(len(bidResp.SeatBid.Bid))`
would be an accurate way to set the capacity.